modules/geo/src/main/java/org/opensearch/geo/search/aggregations/bucket/geogrid/InternalGeoHashGridBucket.java

```diff
@@ -51,8 +52,8 @@ class InternalGeoHashGridBucket extends BaseGeoGridBucket<InternalGeoHashGridBuc
     /**
      * Read from a stream.
      */
-    InternalGeoHashGridBucket(StreamInput in) throws IOException {
-        super(in);
+    InternalGeoHashGridBucket(BaseStreamInput in) throws IOException {
```

Comment:
I believe it is better to keep `StreamInput` (and change to `BaseStreamInput` later on)

Reply:
IIRC, this fails the build w/ an invalid ctor when `GeoHashGrid` calls `InternalGeohashGridBucket::new`

Reply:
That's because the raw types (Reader), with correct type parameters build is green:

```
    @Override
    protected Reader<InternalGeoHashGridBucket> getBucketReader() {
        return InternalGeoHashGridBucket::new;
    }

```

Reply:
Right, that's what I was referring to. This change is made in the followup PR where I'm pulling up the primitive reader/writers, but (to your point) the change surface area is minimized if I do this part now. So I went ahead and took care of it in this PR.